# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/ShadowRoot.h
# Commit: f9cf0740eef2
# Full Hash: f9cf0740eef2fcae3e2f93afe49df1569bd89c03
# Author: Adam Vandolder <avandolder@mozilla.com>
# Date: 2024-05-15 20:24:18
# Regressor Bug: 1890428
# File Overlap Count: 1
# Description:
#   Bug 1890428 - Add serialization support for Declarative Shadow DOM. r=webidl,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D206955
# ==============================================================================

diff -r 37e2a10a9022 -r f9cf0740eef2 dom/base/ShadowRoot.h
--- a/dom/base/ShadowRoot.h	Tue May 14 16:43:27 2024 +0000
+++ b/dom/base/ShadowRoot.h	Tue May 14 16:43:28 2024 +0000
@@ -44,6 +44,7 @@
 
   using Declarative = Element::ShadowRootDeclarative;
   using IsClonable = Element::ShadowRootClonable;
+  using IsSerializable = Element::ShadowRootSerializable;
 
  public:
   NS_IMPL_FROMNODE_HELPER(ShadowRoot, IsShadowRoot());
@@ -54,7 +55,7 @@
   ShadowRoot(Element* aElement, ShadowRootMode aMode,
              Element::DelegatesFocus aDelegatesFocus,
              SlotAssignmentMode aSlotAssignment, IsClonable aClonable,
-             Declarative aDeclarative,
+             IsSerializable aIsSerializable, Declarative aDeclarative,
              already_AddRefed<mozilla::dom::NodeInfo>&& aNodeInfo);
 
   void AddSizeOfExcludingThis(nsWindowSizes&, size_t* aNodeSize) const final;
@@ -83,6 +84,7 @@
   SlotAssignmentMode SlotAssignment() const { return mSlotAssignment; }
   bool Clonable() const { return mIsClonable == IsClonable::Yes; }
   bool IsClosed() const { return mMode == ShadowRootMode::Closed; }
+  bool Serializable() const { return mIsSerializable == IsSerializable::Yes; }
 
   void RemoveSheetFromStyles(StyleSheet&);
   void RuleAdded(StyleSheet&, css::Rule&);
@@ -247,6 +249,8 @@
   MOZ_CAN_RUN_SCRIPT
   void SetHTMLUnsafe(const nsAString& aHTML);
 
+  void GetHTML(const GetHTMLOptions& aOptions, nsAString& aResult);
+
  protected:
   // FIXME(emilio): This will need to become more fine-grained.
   void ApplicableRulesChanged();
@@ -290,6 +294,9 @@
   // https://dom.spec.whatwg.org/#shadowroot-clonable
   const IsClonable mIsClonable;
 
+  // https://dom.spec.whatwg.org/#shadowroot-serializable
+  const IsSerializable mIsSerializable;
+
   nsresult Clone(dom::NodeInfo*, nsINode** aResult) const override;
 };
 